test/it/uk/gov/hmrc/component/CaseSpec.scala

```diff
@@ -469,11 +513,158 @@ class CaseSpec extends BaseFeatureSpec {
+
+      storeCases(c1, c2, c7)
+
+      val result = Http(s"$serviceUrl/cases?good_description=laptop").asString
+
+      result.code shouldEqual OK
+      Json.parse(result.body) shouldBe Json.toJson(Seq(c7))
+    }
+
+    scenario("Filtering by substring") {
+
+      storeCases(c1, c2, c7, c8)
+
+      val result = Http(s"$serviceUrl/cases?good_description=laptop").asString
+
+      result.code shouldEqual OK
+      Json.parse(result.body) shouldBe Json.toJson(Seq(c7, c8))
+    }
+  }
+
+
+  feature("Get Cases by multiple parameters") {
+    // TODO
+  }
+
+
+  feature("Get Cases sorted by commodity code") {
+
+    val caseWithEmptyCommCode = createCase().copy(decision = None)
+    val caseY1 = createCase().copy(decision = Some(createDecision(bindingCommodityCode = "777")))
+    val caseY2 = createCase().copy(decision = Some(createDecision(bindingCommodityCode = "777")))
+    val caseZ = createCase().copy(decision = Some(createDecision(bindingCommodityCode = "1111111111")))
+
+    scenario("Sorting default - ascending order") {
+      Given("There are few cases in the database")
+      storeCases(caseY2, caseWithEmptyCommCode, caseY1, caseZ)
+
+      When("I get all cases sorted by commodity code")
+      val result = Http(s"$serviceUrl/cases?sort_by=commodity-code").asString
+
+      Then("The response code should be 200")
+      result.code shouldEqual OK
+
+      And("The expected cases are returned in the JSON response")
+      Json.parse(result.body) shouldBe Json.toJson(Seq(caseWithEmptyCommCode, caseZ, caseY2, caseY1))
+    }
+
+    scenario("Sorting in ascending order") {
+      Given("There are few cases in the database")
+      storeCases(caseY1, caseWithEmptyCommCode, caseY2, caseZ)
+
+      When("I get all cases sorted by commodity code")
+      val result = Http(s"$serviceUrl/cases?sort_by=commodity-code&sort_direction=asc").asString
+
+      Then("The response code should be 200")
+      result.code shouldEqual OK
+
+      And("The expected cases are returned in the JSON response")
+      Json.parse(result.body) shouldBe Json.toJson(Seq(caseWithEmptyCommCode, caseZ, caseY1, caseY2))
+    }
+
+    scenario("Sorting in descending order") {
+      Given("There are few cases in the database")
+      storeCases(caseZ, caseWithEmptyCommCode, caseY1, caseY2)
+
+      When("I get all cases sorted by commodity code")
+      val result = Http(s"$serviceUrl/cases?sort_by=commodity-code&sort_direction=desc").asString
+
+      Then("The response code should be 200")
+      result.code shouldEqual OK
+
+      And("The expected cases are returned in the JSON response")
+      Json.parse(result.body) shouldBe Json.toJson(Seq(caseY1, caseY2, caseZ, caseWithEmptyCommCode))
+    }
+
+  }
+
+
   feature("Get Cases sorted by days elapsed") {
-    val oldCase = c1.copy(daysElapsed = 1)
-    val newCase = c2.copy(daysElapsed = 0)
 
-    scenario("Sorting default") {
+    val oldCase = c1.copy(daysElapsed = 0)
+    val newCase = c2.copy(daysElapsed = 1)
```

Comment:
technically the higher the daysElapsed, the older the case so it made more sense before